### ISSUE TITLE
block-scoped-var false positive on destructuring in function argument

### ISSUE THREAD

**Issue by olalonde** [eslint/eslint#3124]
I am refactoring an Express app to es6 and tried to do this:

```javascript
module.exports = {
  index: ({
    app,
    query: { limit = 10, page = 1, type },
    params: { param1, param2 }
  }, res, next) => {
     // some code
  },
};
```

It runs and compiles as expected using babel. However, eslint gives me a bunch of `block-scoped-var` errors:

```
"limit" used outside of binding context. (block-scoped-var)
etc.
```

Is that a bug or am I doing something wrong?

Also posted on Airbnb style guide repo as it might be a problem with their `.eslintrc` file. (https://github.com/airbnb/javascript/issues/434)

`v0.24.1`

**Comment by eslintbot** [comment]
Thanks for the issue! We get a lot of issues, so this message is automatically posted to each one to help you check that you've included all of the information we need to help you.

Reporting a bug? Please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
1. The source code that caused the problem
1. The configuration you're using (for the rule or your entire config file)
1. The actual ESLint output complete with line numbers

Requesting a new rule? Please be sure to include:

1. The use case for the rule - what is it trying to prevent or flag?
1. Whether the rule is trying to prevent an error or is purely stylistic
1. Why you believe this rule is generic enough to be included

Requesting a feature? Please be sure to include:

1. The problem you want to solve (don't mention the solution)
1. Your take on the correct solution to problem

Including this information in your issue helps us to triage it and get you a response as quickly as possible.

Thanks!

**Comment by mysticatea** [comment]
Thank you for the report.

This is probably duplicate of #2253.
I'm fixing just now :)

**Comment by olalonde** [comment]
Oh yeah, seems like duplicate. Will follow that issue.
